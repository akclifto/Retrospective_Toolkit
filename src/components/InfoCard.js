import React from "react";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import PropTypes from "prop-types";
import "../styles/InfoCard.css";

function SetInfoCard(props) {
  const { body, body2, body3 } = props;

  return (
    <Card className="infoCard--root" variant="outlined">
      <CardContent>
        <div>
          <Typography variant="body1" component="p">
            {body}
          </Typography>
          <Typography component="p">
            <br />
            {body2}
          </Typography>
          <Typography variant="body2" component="p" color="textSecondary">
            {body3}
          </Typography>
        </div>
      </CardContent>
      <CardActions />
    </Card>
  );
}

class InfoCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = { showActivityInfo: false };
    // eslint rules force og constructor, bind the functions
    this.clickHandler = this.clickHandler.bind(this);
  }

  clickHandler = () => {
    this.setState((state) => ({ showActivityInfo: !state.showActivityInfo }));
  };

  render() {
    const { showActivityInfo } = this.state;
    const { title, body, body2, body3 } = this.props;

    return (
      <div className="container">
        <button
          className="title-button"
          type="button"
          onClick={this.clickHandler}
        >
          <Typography
            variant="h4"
            color="textPrimary"
            gutterBottom
            align="center"
          >
            {title}
          </Typography>
        </button>
        {showActivityInfo ? (
          <div className="info-open">
            <SetInfoCard body={body} body2={body2} body3={body3} />
          </div>
        ) : (
          <div className="info-close" />
        )}
      </div>
    );
  }
}

SetInfoCard.propTypes = {
  body: PropTypes.string.isRequired,
  body2: PropTypes.string.isRequired,
  body3: PropTypes.string.isRequired,
};

InfoCard.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  body2: PropTypes.string.isRequired,
  body3: PropTypes.string.isRequired,
};

export default InfoCard;
