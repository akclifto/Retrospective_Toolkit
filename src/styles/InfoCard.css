.infoCard--root {
  width: 96.6vw;
  margin: auto;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
  overflow: hidden;
}

.container {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

.title-button {
  width: 96.6vw;
  padding-top: 12px; /* top right bottom left */
  margin-top: 12px;
  text-align: center;
  background-color: white;
  border: 1px solid rgba(0, 0, 0, 0.12);
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
}

.title-button:hover {
  box-shadow: 0 10px 13px rgba(0, 0, 0, 0.25);
  background-color: rgba(246, 248, 250, 0.356);
}

.title-button:focus {
  border: none;
  text-decoration: none;
  box-shadow: none;
}

.info-open {
  max-height: 400px;
  transition: max-height 0.4s ease;
  height: 88%;
}

.info-close {
  max-height: 0px;
  transition: max-height 0.4s ease;
}
