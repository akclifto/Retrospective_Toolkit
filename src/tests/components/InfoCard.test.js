/* eslint-disable no-console */
import React from "react";
import { shallow } from "enzyme";
import InfoCard from "../../components/InfoCard";

let wrapper;

beforeEach(() => {
  jest.spyOn(console, "warn").mockImplementation(() => {});
  jest.spyOn(console, "error").mockImplementation(() => {});
  jest.spyOn(console, "log").mockImplementation(() => {});
});

describe("Components/InfoCard Testing", () => {
  // Snapshot for InfoCard
  test("SNAPSHOT: InfoCard", () => {
    wrapper = shallow(<InfoCard />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Test clickHandler event", () => {
    wrapper = shallow(<InfoCard clickHander={jest.fn()} />);
    wrapper.find("button").simulate("click");
    expect(wrapper).toMatchSnapshot();
  });
});
