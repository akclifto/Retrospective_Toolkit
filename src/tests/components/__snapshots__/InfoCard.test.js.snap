// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/InfoCard Testing SNAPSHOT: InfoCard 1`] = `
<div
  className="container"
>
  <button
    className="title-button"
    onClick={[Function]}
    type="button"
  >
    <WithStyles(ForwardRef(Typography))
      align="center"
      color="textPrimary"
      gutterBottom={true}
      variant="h4"
    />
  </button>
  <div
    className="info-close"
  />
</div>
`;

exports[`Components/InfoCard Testing Test clickHandler event 1`] = `
<div
  className="container"
>
  <button
    className="title-button"
    onClick={[Function]}
    type="button"
  >
    <WithStyles(ForwardRef(Typography))
      align="center"
      color="textPrimary"
      gutterBottom={true}
      variant="h4"
    />
  </button>
  <div
    className="info-open"
  >
    <SetInfoCard />
  </div>
</div>
`;
